examples/chat/server/server.bal

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) 2021 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/io;
+import ballerina/websocket;
+
+final string USERNAME = "username";
+map<websocket:Caller> connectionsMap = {};
+
+service /chat on new websocket:Listener(9090) {
+    resource function get [string username]() returns websocket:Service|websocket:UpgradeError {
+        if (username != "") {
```

Comment:
How could the username be "" here ? What would be the sample URL for this scenario ?

Reply:
Username cannot be `""`here. It is a mistake. Moved this validation to the client-side.